### PR TITLE
Add python `slice` API.

### PR DESCRIPTION
**Pull request by kevinstephano**
This PR adds the `slice` operation as is documented by the [XLA Slice Op](https://www.tensorflow.org/xla/operation_semantics#slice).

### CODE REVIEW

File: python_tests/test_python_frontend.py [hunk 1]
```diff
@@ -1468,5 +1468,25 @@ def fusion_func(fd: FusionDefinition) :
...
+            torch.randn(1, 1, 1024, 1204, device='cuda'),
+            torch.randn(32, 16, 128, 128, device='cuda'),
+        ]
+
+        def nvfuser_fusion_id(fd : FusionDefinition) -> None :
+            t0 = fd.from_pytorch(inputs[0])
+            t1 = fd.from_pytorch(inputs[1])
+            t1_sizes = fd.ops.tensor_sizes(t1)
+            t2 = fd.ops.slice(t0, [0, 0, 0, 0], [1, 1, 128, 128], [1, 1, 1, 1])
+            t3 = fd.ops.broadcast_in_dim(t2, output_shape=t1_sizes, broadcast_dims=[0, 1, 2, 3])
```
Comment (jjsjann123) [inline python_tests/test_python_frontend.py:1482]: I can never remember what broadcast_in_dim means... :laughing:

**Review by jjsjann123 — APPROVED** [review]
LGTM

**Comment by jjsjann123** [comment]
I would nitpick on our version number. But I think it doesn't really matter until our user start having release code.

**Comment by csarofeen** [comment]
Makes sense to get in the habit of incrementing the version.

**Comment by kevinstephano** [comment]
> I would nitpick on our version number. But I think it doesn't really matter until our user start having release code.

I bumped the version.

**Pull request merged**